container/lxd/shared.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Licensed under the Apache 2.0 license. See github.com/lxc/lxd COPYING file for details.
+
+package lxd
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"os/user"
+	"time"
+)
+
+// The following file exists because we can no longer import the
+// github.com/canonical/lxd/shared package when we build with CGO_ENABLED=1.
```

Comment:
What's our strategy for keeping this vendored code up to date with any upstream changes?

Reply:
This was backported from 3.5/3.4 so that we could update the dependency.